src/deepsparse/pipeline.py

```diff
@@ -638,6 +661,9 @@ def _run_with_static_batch(self, pipeline_inputs: BaseModel) -> BaseModel:
                 f"Outputs of {self.__class__} must be instances of "
                 f"{self.output_schema} found output of type {type(pipeline_outputs)}"
             )
+        self.timer.post_process_complete()
+
+        self.inference_timing = self.timer.build()
```

Comment:
How does this stuff interact with dynamic batch which uses async?

Perhaps _run_with_static batch should create & return a timer inside of it, and _run_with_dynamic_batch can sum them all together or take max or something else

Reply:
I think the way to do this is to gather the time of the start of the first dynamic inference and and the end of the last dynamic inference. This is how `deepsparse.benchmark` tracks time. Computing this from just deltas seems difficult.

Reply:
@mgoin I would lean towards @corey-nm suggestion. The recent commit collects all the separate inference timing readouts in an `Iterable[InferenceTimingSchema]` and sums over all the labels. I like it because :
1. we are not losing any information (e.g. we can still see roughly how much time the input spent in pre-processing step alone)
2. `total_time` still is equal to the sum of the `pre_processing`, `post_processing` and `engine_forward`